### PR TITLE
[Microsoft.BillingBenefits] Initializing new service

### PR DESCRIPTION
**Pull request by gaoyp830**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by Yao725 — COMMENTED** [review]
Hi @gaoyp830, is this namespace approved by the arch board? If not, you can open an issue (example: https://github.com/Azure/azure-sdk/issues/5026) in [azure-sdk](https://github.com/Azure/azure-sdk) to trigger the namespace review process.

**Comment by gaoyp830** [comment]
> Hi @gaoyp830, is this namespace approved by the arch board? If not, you can open an issue (example: [Azure/azure-sdk#5026](https://github.com/Azure/azure-sdk/issues/5026)) in [azure-sdk](https://github.com/Azure/azure-sdk) to trigger the namespace review process.

Created request: https://github.com/Azure/azure-sdk/issues/5181

**Comment by gaoyp830** [comment]
> Hi @gaoyp830, is this namespace approved by the arch board? If not, you can open an issue (example: [Azure/azure-sdk#5026](https://github.com/Azure/azure-sdk/issues/5026)) in [azure-sdk](https://github.com/Azure/azure-sdk) to trigger the namespace review process.

@Yao725  I have got namespace approval for .net https://github.com/Azure/azure-sdk/issues/5181, please continue reviewing this PR

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure BillingBenefits management client library for .NET
+
+Microsoft BillingBenefits is a Microsoft-managed service which allows management on Azure savings plan. See here for more details about savings plan: https://learn.microsoft.com/en-us/azure/cost-management-billing/savings-plan/savings-plan-compute-overview
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/README.md:3]: Please remove the locale information of the link.

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+using NUnit.Framework;
+using Azure.ResourceManager.BillingBenefits.Models;
+using Azure.Core;
+
+namespace Azure.ResourceManager.BillingBenefits.Tests
+{
+    public class ReservationOrderAliasTests : BillingBenefitsManagementTestBase
+    {
+        private ReservationOrderAliasModelResource ModelResource { get; set; }
+        private TenantResource tenantResource { get; set; }
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs:18]: I think it's better to use field here.
```suggestion
        private ReservationOrderAliasModelResource _modelResource;
        private TenantResource _tenantResource;
```

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            Assert.AreEqual("Standard_B1s", Data.SkuName);
+            Assert.AreEqual(Term.P3Y, Data.Term);
+            Assert.AreEqual(1, Data.Quantity);
+            Assert.AreEqual(AzureLocation.EastUS, Data.Location);
+            Assert.AreEqual(ReservedResourceType.VirtualMachines, Data.ReservedResourceType);
+            Assert.AreEqual(InstanceFlexibility.On, Data.ReservedResourceInstanceFlexibility);
+            Assert.False(Data.Renew);
+        }
+
+        private static ReservationOrderAliasModelCreateOrUpdateContent CreatePurchaseRequestContent(AppliedScopeType scope)
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs:184]: Maybe it's better to move this method to `TestHelpers`?

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            request.DisplayName = "TestROName" + scope.ToString();
+            request.BillingPlan = new BillingPlan("P1M");
+            request.ReservedResourceType = "VirtualMachines";
+            request.Renew = false;
+            request.Quantity = 1;
+            request.Location = AzureLocation.EastUS;
+            request.ReservedResourceProperties = new ReservationOrderAliasRequestReservedResourceProperties
+            {
+                InstanceFlexibility = InstanceFlexibility.On
+            };
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/ReservationOrderAliasTests.cs:199]:
```suggestion
            var request = new ReservationOrderAliasModelCreateOrUpdateContent(new BillingBenefitsSku("Standard_B1s"))
            {
                BillingScopeId = "/subscriptions/eef82110-c91b-4395-9420-fcfcbefc5a47",
                Term = new Term("P3Y"),
                AppliedScopeType = scope,
                DisplayName = "TestROName" + scope.ToString(),
                BillingPlan = new BillingPlan("P1M"),
                ReservedResourceType = "VirtualMachines",
                Renew = false,
                Quantity = 1,
                Location = AzureLocation.EastUS,
                ReservedResourceProperties = new ReservationOrderAliasRequestReservedResourceProperties
                {
                    InstanceFlexibility = InstanceFlexibility.On
                }
            };
```

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/SavingsPlansOrderAliasTests.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+using NUnit.Framework;
+using Azure.ResourceManager.BillingBenefits.Models;
+using Azure.ResourceManager.BillingBenefits.Tests.Helper;
+
+namespace Azure.ResourceManager.BillingBenefits.Tests
+{
+    public class SavingsPlansOrderAliasTests : BillingBenefitsManagementTestBase
+    {
+        private SavingsPlanOrderAliasModelResource ModelResource { get; set; }
+        private TenantResource tenantResource { get; set; }
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/SavingsPlansOrderAliasTests.cs:18]:
```suggestion
        private SavingsPlanOrderAliasModelResource _modelResource;
        private TenantResource _tenantResource;
```

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/SavingsPlansOrdersTests.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.BillingBenefits.Tests
+{
+    public class SavingsPlansOrdersTests : BillingBenefitsManagementTestBase
+    {
+        private TenantResource Tenant { get; set; }
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/Scenario/SavingsPlansOrdersTests.cs:15]:
```suggestion
        private TenantResource _tenant;
```

File: sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/TestHelpers.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            request.Term = new Term("P3Y");
+            request.AppliedScopeType = scope;
+            request.DisplayName = "TestSPName" + scope.ToString();
+            request.BillingPlan = new BillingPlan("P1M");
+            request.Commitment = new Commitment
+            {
+                Grain = "Hourly",
+                CurrencyCode = "USD",
+                Amount = 0.001
+            };
```
Comment (Yao725) [inline sdk/billingbenefits/Azure.ResourceManager.BillingBenefits/tests/TestHelpers.cs:24]:
```suggestion
            var request = new SavingsPlanOrderAliasModelData(new BillingBenefitsSku("Compute_Savings_Plan"))
            {
                BillingScopeId = "/subscriptions/eef82110-c91b-4395-9420-fcfcbefc5a47",
                Term = new Term("P3Y"),
                AppliedScopeType = scope,
                DisplayName = "TestSPName" + scope.ToString(),
                BillingPlan = new BillingPlan("P1M"),
                Commitment = new Commitment
                {
                    Grain = "Hourly",
                    CurrencyCode = "USD",
                    Amount = 0.001
                }
            };
```

**Review by Yao725 — COMMENTED** [review]
Please also fix the spell error in CI, you can refer to [this](https://github.com/HarveyLink/azure-sdk-for-net/blob/375265eb56936d43218f902bfb7727027b814129/.vscode/cspell.json) as example.

**Comment by gaoyp830** [comment]
@Yao725 please approve and merge the PR to generate beta SDK if it looks good to you.

**Review by Yao725 — APPROVED** [review]

**Pull request merged**